File: instantlib/src/main/java/com/baselib/instant/thread/NamedTask.java
package com.baselib.instant.thread;

/**
 * 拥有名称的Runnable
 * @author wsb
 */
public class NamedTask implements Runnable {

    public Runnable mTask;
    public int mPriority = Thread.NORM_PRIORITY;
    public String mThreadName;

    public NamedTask(Runnable task) {
        mTask = task;
    }

    @Override
    public void run() {
        mTask.run();
    }
}

File: instantlib/src/main/java/com/baselib/instant/thread/NamedTask.kt
package com.baselib.instant.thread

/**
 * 拥有名称的Runnable
 * @author wsb
 */
class NamedTask constructor(private val task: Runnable) : Runnable {

    var threadName: String = ""

    var priority: Int = Thread.NORM_PRIORITY

    override fun run() {
        task.run()
    }

}